core/src/main/java/io/ddf/content/ViewHandler.java

```diff
@@ -137,13 +137,8 @@ public DDF removeColumns(List<String> columnNames) throws DDFException {
     }
 
     DDF newddf = this.project(columns);
-    if(this.getDDF().isMutable()) {
-      this.getDDF().updateInplace(newddf);
-      return this.getDDF();
-    } else {
-      newddf.getMetaDataHandler().copyFactor(this.getDDF());
-      return newddf;
-    }
+    newddf.getMetaDataHandler().copyFactor(this.getDDF());
```

Comment:
not sure what do you mean, of course this will break current behaviour, and we need to change the client also